compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor.hs

```diff
@@ -117,6 +119,30 @@ checkImports x =
     [ (ss, "Import of internal module " ++ GHC.moduleNameString m ++ " is not allowed.")
     | GHC.L ss GHC.ImportDecl{ideclName=GHC.L _ m} <- GHC.hsmodImports $ GHC.unLoc x, isInternal m]
 
+-- | Emit a warning if a record constructor name does not match the record type name.
+-- See issue #4718. This ought to be moved into 'checkDataTypes' before too long.
+checkRecordConstructor :: GHC.ParsedSource -> [(GHC.SrcSpan, String)]
+checkRecordConstructor (GHC.L _ m) = mapMaybe getRecordError (GHC.hsmodDecls m)
+  where
+    getRecordError :: GHC.LHsDecl GHC.GhcPs -> Maybe (GHC.SrcSpan, String)
+    getRecordError (GHC.L ss decl)
+        | GHC.TyClD _ GHC.DataDecl{tcdLName=ltyName, tcdDataDefn=dataDefn} <- decl
+        , GHC.HsDataDefn{dd_cons=[con]} <- dataDefn
+        , GHC.RecCon{} <- GHC.con_args (GHC.unLoc con)
+        , GHC.L _ tyName <- ltyName
+        , [GHC.L _ conName] <- GHC.getConNames (GHC.unLoc con)
+        , tyNameStr <- GHC.occNameString (GHC.rdrNameOcc tyName)
+        , conNameStr <- GHC.occNameString (GHC.rdrNameOcc conName)
```

Comment:
Since there's no pattern matching going on here, I'd suggest to use `let` bindings.
```suggestion
        , let tyNameStr = GHC.occNameString (GHC.rdrNameOcc tyName)
        , let conNameStr = GHC.occNameString (GHC.rdrNameOcc conName)
```

Reply:
Changed it in the other PR.